my-app/src/components/products.js

```diff
@@ -0,0 +1,156 @@
+import { useEffect } from "react";
+import axios from "axios";
+import {
+   useProducts,
+   getSortedData,
+   getPriceRangeData,
+} from "../contexts/ProductContext";
+import "./products.css";
+import { FaShoppingCart } from "react-icons/fa";
+import { postRequestCart, postRequestWishlist } from "./utilities/utilities";
+import {
+   BrowserRouter as Router,
+   Routes,
+   Link,
+   Route,
+   Navigate,
+   useNavigate,
+   useParams,
+   useLocation,
+} from "react-router-dom";
+export function ProductListing() {
+   const { state, dispatch } = useProducts();
+
+   useEffect(() => {
+      (async function () {
+         try {
+            const productData = await axios.get(
+               "https://myfeline-restapi.sohamparab13.repl.co/products"
+            );
+
+            dispatch({
+               type: "SET PRODUCTS",
+               payload: productData.data,
+            });
+         } catch (error) {
+            dispatch({ type: "LOADING", payload: false });
+         }
+      })();
+   }, []);
+
+   let sortedData = getSortedData(state.products, state.sortBy);
+
+   let priceRangeData = getPriceRangeData(sortedData, state.price_range);
+
+   return (
+      <div className="productDiv">
+         <div className="aside">
```

Comment:
You can extract the side bar for filter to a different component and import it here.
It would look super clean if you could separate `productCard`, and `sideDrawer` as their own components and put them together in `productListing.js` page.